src/lib/parts.ts

```diff
@@ -245,18 +320,26 @@ export class NodePart implements Part {
   private __commitText(value: unknown): void {
     const node = this.startNode.nextSibling!;
     value = value == null ? '' : value;
-    // If `value` isn't already a string, we explicitly convert it here in case
-    // it can't be implicitly converted - i.e. it's a symbol.
-    const valueAsString: string =
-        typeof value === 'string' ? value : String(value);
     if (node === this.endNode.previousSibling &&
         node.nodeType === 3 /* Node.TEXT_NODE */) {
       // If we only have a single text node between the markers, we can just
       // set its value, rather than replacing it.
-      // TODO(justinfagnani): Can we just check if this.value is primitive?
-      (node as Text).data = valueAsString;
+      if (sanitizeDOMValue) {
+        value = sanitizeDOMValue(value, 'data', 'property', node);
+      }
+      (node as Text).data = typeof value === 'string' ? value : String(value);
     } else {
-      this.__commitNode(document.createTextNode(valueAsString));
+      // When setting text content, for security purposes it matters a lot what
+      // the parent is. For example, <style> and <script> need to be handled
+      // with care, while <span> does not. So first we need to put a text node
+      // into the document, then we can sanitize its contentx.
+      const textNode = document.createTextNode('');
+      this.__commitNode(textNode);
+      if (sanitizeDOMValue) {
+        value = String(
```

Comment:
Note that by mutating the `value` variable here, we're going to save the sanitized value as the previous value: 

https://github.com/Polymer/lit-html/pull/750/files#diff-63a9df26ecde05415a902b5b5dc5e266R344

This means when a sanitizer changes a value that we'll always re-render this part. Can we say that sanitizers have to be pure functions so we can remember the pre-sanitized value and skip the whole operation if the input value is the same?

Reply:
Good point, that's a reasonable assumption. Separated out the rendered value from `this.value`